optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -121,7 +121,11 @@
 
     public static <Solution_> SolutionDescriptor<Solution_> buildSolutionDescriptor(Class<Solution_> solutionClass,
             List<Class<?>> entityClassList) {
-        return buildSolutionDescriptor(DomainAccessType.REFLECTION, solutionClass, entityClassList);
+        // Auto-detect if we should be using Gizmo, as the user did not specify a preference.
+        DomainAccessType domainAccessType =
+                (ConfigUtils.isInQuarkus() && ConfigUtils.isGizmoOnClasspath()) ? DomainAccessType.GIZMO
+                        : DomainAccessType.REFLECTION;
+        return buildSolutionDescriptor(domainAccessType, solutionClass, entityClassList);
     }
```

Comment:
In tests, I want to "do the right thing" and use the best member accessor that is available.
That requires me to know, at this stage, if I'm running in Quarkus.

This change is controversial. We can just say "it always uses reflection", as it did before. I am OK with that.